### PR TITLE
refactor + fixes / feature additions

### PR DESCRIPTION
**Pull request by TakaRikka**
big refactor of the codebase to make things more readable, consistent, and easier to use. overall memory usage cut down by a decent chunk.

fixes:
- pointer watches dont crash when pointer is null
- timers work correctly now
- norgor save correctly has lent bomb bag
- coro TD checker works correctly
- jumping to memory editor from a pointer watch will jump to offset address

additions:
- frame advance tool
- All Dungeons / Any% saves updated
- link vertical aim angle added to link debug info
- time of day added to link debug info
- heap debug viewer
- actor list menu (displays info from the actor queue)
- region flags added to portal flags menu

### CODE REVIEW

File: include/menu.h [hunk 1]
```diff
@@ -6,21 +6,18 @@
...
 #ifdef GCN_PLATFORM
-#define SELECTION_BUTTON (Controller::Pad::A)
-#define BACK_BUTTON (Controller::B)
+#define SELECTION_BUTTON (GZPad::A)
+#define BACK_BUTTON (GZPad::B)
 #endif
 #ifdef WII_PLATFORM
 #define SELECTION_BUTTON (Controller::Mote::A)
-#define BACK_BUTTON (Controller::B)
+#define BACK_BUTTON (GZPad::B)
```
Comment (kipcode66) [inline include/menu.h:20]: Is GZPad::B equivalent for Wii and GC?

Comment (TakaRikka) [reply include/menu.h:20]: should be fixed with libtp update

**Comment by kipcode66** [comment]
Once the format check works, it looks good to me

**Review by kipcode66 — APPROVED** [review]
lgtm

**Pull request merged**